### PR TITLE
axis mode and chain ltr fixes

### PR DESCRIPTION
**Pull request by jafu888**

### CODE REVIEW

File: projects/MotionLayoutVerification/app/src/main/res/layout/verification_044.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (oscar-ad) [inline projects/MotionLayoutVerification/app/src/main/res/layout/verification_044.xml:2]: These XML files don't seem to do anything.

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/helpers/HorizontalChainReference.java [hunk 2]
```diff
@@ -51,21 +51,31 @@ public void apply() {
...
                 String refKey = reference.getKey().toString();
-                previous.endToStart(reference.getKey()).margin(getPostMargin(preKey));
-                reference.startToEnd(previous.getKey()).margin(getPreMargin(refKey));
+                if (lrMode) {
+                    previous.rightToLeft(reference.getKey()).margin(getPostMargin(preKey));
+                    reference.leftToRight(previous.getKey()).margin(getPreMargin(refKey));
+                } else {
+                    previous.endToStart(reference.getKey()).margin(getPostMargin(preKey));
+                    reference.startToEnd(previous.getKey()).margin(getPreMargin(refKey));
+                }
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/helpers/HorizontalChainReference.java:78]: I suppose this isn't wrong, but as it is, it makes no difference.

`startToStart` will always apply `leftToLeft` constraints.

File: projects/MotionLayoutVerification/app/src/main/res/layout/verification_044.xml (hunk 1 above)
Comment (jafu888) [reply projects/MotionLayoutVerification/app/src/main/res/layout/verification_044.xml]: removed

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/helpers/HorizontalChainReference.java (hunk 2 above)
Comment (jafu888) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/helpers/HorizontalChainReference.java:78]: When we implement left,right,start and end this is the correct behavior

**Review by oscar-ad — APPROVED** [review]

**Pull request merged**